app/javascript/__tests__/Articles/ArticlesDetails.test.tsx

```diff
@@ -1,29 +1,127 @@
-import ArticleDetails from '../../components/Articles/ArticlesDetails';
+import React from "react";
+import "@testing-library/jest-dom/extend-expect";
+import { render } from "@testing-library/react";
+import ArticleDetails from "../../components/Articles/ArticlesDetails";
 
-let container = null;
+const expectedTitle = "mechanism momentum museum minimum mushroom";
+const expectedContent =
+  "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.";
+const expectedFirstName = "Megan";
+const expectedLastName = "Duffy";
+const expectedLink = "/articles";
+const expectedLinkText = "View All Articles";
+const expectedCreatedAt = "27 March 2020";
+const expectedEditLink = "/articles/3/edit";
+const expectedCurrentUser = 3;
+const expectedEditLinkText = "Edit Article";
 
-beforeEach(() => {
-  container = document.createElement("div");
-  document.body.appendChild(container);
-});
+describe("ArticleDetails", () => {
+  describe("Snapshots", () => {
+    test("should render the Article Details when logged out", () => {
+      const { container } = render(
+        <ArticleDetails
+          title={expectedTitle}
+          content={expectedContent}
+          firstName={expectedFirstName}
+          lastName={expectedLastName}
+          link={expectedLink}
+          linkText={expectedLinkText}
+          createdAt={expectedCreatedAt}
+        />
+      );
+      expect(container).toMatchSnapshot();
+    });
 
-afterEach(() => {
-  unmountComponentAtNode(container);
-  container.remove();
-  container = null;
-});
+    test("should render the Article Details when logged in", () => {
+      const { container } = render(
+        <ArticleDetails
+          title={expectedTitle}
+          content={expectedContent}
+          firstName={expectedFirstName}
+          lastName={expectedLastName}
+          link={expectedLink}
+          linkText={expectedLinkText}
+          createdAt={expectedCreatedAt}
+          editLink={expectedEditLink}
+          currentUser={expectedCurrentUser}
+        />
+      );
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe("Props", () => {
+    test("should render the text passed in", () => {
+      const { container } = render(
+        <ArticleDetails
+          title={expectedTitle}
+          content={expectedContent}
+          firstName={expectedFirstName}
+          lastName={expectedLastName}
+          link={expectedLink}
+          linkText={expectedLinkText}
+          createdAt={expectedCreatedAt}
+        />
+      );
+      expect(container).toHaveTextContent(expectedTitle);
```

Comment:
Not essential, but you could be more specific about where specific pieces of text should appear by using the `container.querySelector` to select where it should be. For instance, `expect(container.querySelector('div.articles-details-info')).toHaveTextContent(expectedFirstName)`

Reply:
Sweet, this is a fantastic way to make my tests more specific! Thank you